cmd/signature-v4-utils.go

```diff
@@ -161,15 +161,7 @@ func extractSignedHeaders(signedHeaders []string, reqHeaders http.Header) (http.
 // Trim leading and trailing spaces and replace sequential spaces with one space, following Trimall()
 // in http://docs.aws.amazon.com/general/latest/gr/sigv4-create-canonical-request.html
 func signV4TrimAll(input string) string {
-	// Remove all whitespaces first
-	cleanWhiteSpaces := func(r rune) rune {
-		switch r {
-		case '\t', '\n', '\u000b', '\r', '\f':
-			return ' '
-		}
-		return r
-	}
-	input = strings.Map(cleanWhiteSpaces, input)
-	// Compress adjacent spaces to one space and return
+	// Compress adjacent spaces (a space is determined by
+	// unicode.IsSpace() internally here) to one space and return
```

Comment:
Sorry to nit-pick but `unicode.IsSpace` actually considers all unicode classified space characters too - the above list is only in the Latin-1 subset - relevant doc snippet:

```
 Other definitions of spacing characters are set by category Z and property Pattern_White_Space. 
```

Still code is simpler, and I don't think it will every break - header fields are only ever expected to be generally compatible when encoded as ASCII (https://stackoverflow.com/questions/4400678/what-character-encoding-should-i-use-for-a-http-header#4410331)